ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/GrpcHealthService.scala

```diff
@@ -8,36 +8,57 @@ import akka.stream.Materializer
 import akka.stream.scaladsl.Source
 import com.digitalasset.grpc.adapter.ExecutionSequencerFactory
 import com.digitalasset.grpc.adapter.utils.DirectExecutionContext
+import com.digitalasset.ledger.api.health.HealthChecks
 import com.digitalasset.platform.api.grpc.GrpcApiService
 import com.digitalasset.platform.server.api.DropRepeated
-import io.grpc.ServerServiceDefinition
+import com.digitalasset.platform.server.api.services.grpc.GrpcHealthService._
 import io.grpc.health.v1.health.{
   HealthAkkaGrpc,
   HealthCheckRequest,
   HealthCheckResponse,
   HealthGrpc
 }
+import io.grpc.{ServerServiceDefinition, Status, StatusException}
 
-import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
 import scala.concurrent.{ExecutionContext, Future}
 
 class GrpcHealthService(
+    healthChecks: HealthChecks,
+    maximumWatchFrequency: FiniteDuration = 1.second,
+)(
     implicit protected val esf: ExecutionSequencerFactory,
     protected val mat: Materializer,
     executionContext: ExecutionContext,
 ) extends HealthAkkaGrpc
     with GrpcApiService {
-  private val servingResponse = HealthCheckResponse(HealthCheckResponse.ServingStatus.SERVING)
-
   override def bindService(): ServerServiceDefinition =
     HealthGrpc.bindService(this, DirectExecutionContext)
 
   override def check(request: HealthCheckRequest): Future[HealthCheckResponse] =
-    Future.successful(servingResponse)
+    Future(matchResponse(Option(request.service).filter(_.nonEmpty)))
 
   override def watchSource(request: HealthCheckRequest): Source[HealthCheckResponse, NotUsed] =
     Source
-      .repeat(servingResponse)
-      .throttle(1, per = 1.second)
+      .fromIterator(() =>
+        Iterator.continually(matchResponse(Option(request.service).filter(_.nonEmpty))))
+      .throttle(1, per = maximumWatchFrequency)
       .via(DropRepeated())
+
+  private def matchResponse(componentName: Option[String]): HealthCheckResponse = {
```

Comment:
It's probably safer to wrap this in a `Try` and wrap it in a `Future` with `Future.fromTry`.

Reply:
I did that and then undid it because too much noise. I think I prefer it though, so I'll change it back.